flang/lib/Lower/FIRBuilder.cpp

```diff
@@ -434,6 +441,51 @@ struct CharacterOpsBuilderImpl {
 
     return {substringRef, substringLen};
   }
+
+  mlir::Value createLenTrim(Char str) {
+    // Note: Runtime for LEN_TRIM should also be available at some
+    // point. For now use an inlined implementation.
+    auto indexType = builder.getIndexType();
+    mlir::Value len = builder.createHere<fir::ConvertOp>(indexType, str.len);
+    auto data = materializeValue(str);
+    auto one = builder.createIntegerConstant(indexType, 1);
+    auto i1Type = mlir::IntegerType::get(1, builder.getContext());
+    auto trueVal = builder.createIntegerConstant(i1Type, 1);
+    auto blank = createBlankConstant(str.getCharacterType());
+    // FIXME: Conversion wrong in general, it would be cool to compare fir.char
+    // without cast. Needs the kindmap.
```

Comment:
Does this mean comparing a single `!fir.char<n>` with another `!fir.char<n>`?

Reply:
Yes, I did not find a way to directly compare two `!fir.char<n>`  without going through the related integer type.  This is now easy to do since you have added the kindMap to the builder.